plugins/google/pkg/googleclient/google.go

```diff
@@ -0,0 +1,178 @@
+package googleclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	admin "google.golang.org/api/admin/directory/v1"
+	"google.golang.org/api/option"
+)
+
+type GoogleClient struct {
+	googleClient *admin.Service
+}
+
+func GetRefreshToken(ctx context.Context, clientID, clientSecret string, port int) (string, error) {
+	redirectURL := fmt.Sprintf("http://localhost:%d", port)
+	// Configure the OAuth 2.0 client credentials
+	config := &oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		RedirectURL:  redirectURL,
+		Scopes: []string{
+			admin.AdminDirectoryUserScope,
+			admin.AdminDirectoryGroupScope,
+		},
+		Endpoint: google.Endpoint,
+	}
+
+	// Create an HTTP server for handling the OAuth 2.0 callback
+	server := &http.Server{Addr: fmt.Sprintf(":%d", port), ReadHeaderTimeout: 5 * time.Second}
+
+	// Create a channel to receive the authorization code
+	authCodeChan := make(chan string)
+
+	// Start the HTTP server in a separate goroutine
+	go func() {
+		http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+			code := r.URL.Query().Get("code")
+			fmt.Fprintf(w, "Authorization code received. You can close this tab now.")
+			authCodeChan <- code
+		})
+
+		if err := server.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			log.Fatalf("Failed to start HTTP server: %v", err)
+		}
+	}()
+
+	// Generate the authorization URL with "access_type=offline" and "prompt=consent"
+	authURL := config.AuthCodeURL("state-token", oauth2.AccessTypeOffline, oauth2.SetAuthURLParam("prompt", "consent"))
+
+	fmt.Printf("Go to the following URL to authorize the application:\n\n%s\n\n", authURL)
+	fmt.Println("Waiting for authorization code...")
+
+	// Receive the authorization code from the channel
+	authCode := <-authCodeChan
+
+	// Exchange the authorization code for an access token
+	token, err := config.Exchange(ctx, authCode)
+	if err != nil {
+		log.Fatalf("Failed to exchange authorization code for access token: %v", err)
```

Comment:
`log.Fatalf` exits the program immediately. Is that really what we want here?
If you just want the output to be written to stderr, you can use `log.Printf` instead.